data_structures/binary_search_tree.cpp

```diff
@@ -122,9 +159,64 @@ void Post(node *n) {
     }
 }
 
+/** test traverse a tree with Post order
+ * and save the traverse node to an array.
+ * \param[in] n the node pointer of a tree to start traverse
+ * \param[out] arr the output array that save the traversed tree nodes pointers
+ */
+void testInOrderTraverse(node *n, std::vector<int> &arr) {
+    if (n != NULL) {
+        testInOrderTraverse(n->left, arr);
+        std::cout << n->val << "  ";
+        arr.push_back(n->val);
+        testInOrderTraverse(n->right, arr);
+    }
+}
+
+/** test function.
+ * auto build a binary tree with special nodes.
+ * test the tree Insert() and Remove() function and out test result.
+ */
+void test_tree() {
+    node *root = new node;
```

Comment:
Dynamic memory allocated here must be freed.

Reply:
yes，fixed

Reply:
Though the applied fix works, can you implement a function to which, a parent node can be passed and it will iteratively free its children? If not, this is still acceptable.

Reply:
Good idea! I  have fixed it.